2024.1/metametamodel/metametamodel.adoc

```diff
@@ -0,0 +1,1145 @@
+|<<Classifier.features>> |Containment |Classifier-features |LionWeb.LionCore_M3.Classifier.features
+|<<Concept>> |Concept |Concept |LionWeb.LionCore_M3.Concept
+|<<Concept.abstract>> |Property |Concept-abstract |LionWeb.LionCore_M3.Concept.abstract
+|<<Concept.extends>> |Reference |Concept-extends |LionWeb.LionCore_M3.Concept.extends
+|<<Concept.implements>> |Reference |Concept-implements |LionWeb.LionCore_M3.Concept.implements
+|<<Containment>> |Concept |Containment |LionWeb.LionCore_M3.Containment
+|<<DataType>> |Concept |DataType |LionWeb.LionCore_M3.DataType
+|<<Enumeration>> |Concept |Enumeration |LionWeb.LionCore_M3.Enumeration
+|<<Enumeration.literals>> |Containment |Enumeration-literals |LionWeb.LionCore_M3.Enumeration.literals
+|<<EnumerationLiteral>> |Concept |EnumerationLiteral |LionWeb.LionCore_M3.EnumerationLiteral
+|<<Feature>> |Concept |Feature |LionWeb.LionCore_M3.Feature
+|<<Feature.optional>> |Property |Feature-optional |LionWeb.LionCore_M3.Feature.optional
+|<<Field>> |Concept |Field |LionWeb.LionCore_M3.Field
+|<<Field.type>> |Reference |Field-type |LionWeb.LionCore_M3.Field.type
+|<<IKeyed>> |Interface |IKeyed |LionWeb.LionCore_M3.IKeyed
+|<<IKeyed.key>> |Property |IKeyed-key |LionWeb.LionCore_M3.IKeyed.key
+|<<Interface>> |Concept |Interface |LionWeb.LionCore_M3.Interface
+|<<Interface.extends>> |Reference |Interface-extends |LionWeb.LionCore_M3.Interface.extends
+|<<Language>> |Concept |Language |LionWeb.LionCore_M3.Language
+|<<Language.dependsOn>> |Reference |Language-dependsOn |LionWeb.LionCore_M3.Language.dependsOn
+|<<Language.entities>> |Containment |Language-entities |LionWeb.LionCore_M3.Language.entities
+|<<Language.version>> |Property |Language-version |LionWeb.LionCore_M3.Language.version
+|<<LanguageEntity>> |Concept |LanguageEntity |LionWeb.LionCore_M3.LanguageEntity
+|<<Link>> |Concept |Link |LionWeb.LionCore_M3.Link
+|<<Link.multiple>> |Property |Link-multiple |LionWeb.LionCore_M3.Link.multiple
+|<<Link.type>> |Reference |Link-type |LionWeb.LionCore_M3.Link.type
+|<<PrimitiveType>> |Concept |PrimitiveType |LionWeb.LionCore_M3.PrimitiveType
+|<<Property>> |Concept |Property |LionWeb.LionCore_M3.Property
+|<<Property.type>> |Reference |Property-type |LionWeb.LionCore_M3.Property.type
+|<<Reference>> |Concept |Reference |LionWeb.LionCore_M3.Reference
+|<<StructuredDataType>> |Concept |StructuredDataType |LionWeb.LionCore_M3.StructuredDataType
+|<<StructuredDataType.fields>> |Containment |StructuredDataType-fields |LionWeb.LionCore_M3.StructuredDataType.fields
+|===
+
+[[predefined-builtins-keys]]
+==== Keys and resolveInfos of built-in elements
+The language hosting built-in elements has id and key `LionCore-builtins`{fn-org141}, its name is `LionCore_builtins`{fn-org195}.
+
+Every language implicitly depend on this language.{fn-org9-implicit}
+Thus, the ids in this language MUST be stable.
+This means the id MUST be identical to the key for each node in this language.
+
```

Comment:
Is there a specific reason why the keys of builtins not similar to the keys in the LionCore_M3 language, namely just their simple name? For example the key of "String" could just be "String", similar to "Concept" having key "Concept".
If there is no specific reason, should we make them similar?

Reply:
In lw2023.1, they were identical to the node's id, and we wanted the id to be as unique as possible.

We don't have the requirement of id == key anymore, but the keys are set now, and should not change ever.